File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.cpp
class Solution {
public:
    int totalNumbers(vector<int>& digits) {
        int n = digits.size();
        int totalEven =0;
        vector<int> digitCount(10,0);

        for(int i =0;i<n;i++){
            digitCount[digits[i]]++;
        }

        for(int i =1;i<=9;i++){
            if(digitCount[i]==0){
                continue;
            }
            digitCount[i]--;
            for(int j =0;j<=9;j++){
                if(digitCount[j]==0){
                    continue;
                }
                digitCount[j]--;

                for(int k =0;k<=8;k+=2){
                    if(digitCount[k]==0){
                        continue;
                    }

                    totalEven++;
                }
                digitCount[j]++;
            }
            digitCount[i]++;

        }
        return totalEven;
    }
};

File: 3483-unique-3-digit-even-numbers/3483-unique-3-digit-even-numbers.py
class Solution:
    def totalNumbers(self, digits: List[int]) -> int:
        n= len(digits)
        totalEven = 0
        digitCount = [0]*10

        for i in range(n):
            digitCount[digits[i]]+=1
        
        for i in range(1,10):
            if digitCount[i]==0:
                continue
            digitCount[i]-=1

            for j in range(10):
                if digitCount[j]==0:
                    continue
                digitCount[j]-=1

                for k in range(0,9,2):
                    if digitCount[k]==0:
                        continue
                    totalEven+=1
                digitCount[j]+=1
            digitCount[i]+=1
        return totalEven